test/functional/api/v2/smart_proxies_controller_test.rb

```diff
@@ -269,5 +269,31 @@ def setup_import_classes
       end
       assert_response :success
     end
+
+    context "proxy api connectivity" do
+      test "should return Proxy's version" do
+        proxy = smart_proxies(:one)
+        fake_version = {:version => '1.11'}
+        ProxyAPI::Version.any_instance.expects(:get).returns(fake_response(fake_version))
```

Comment:
Expect a method at a higher level, e.g. SmartProxy#version and then use unit tests to ensure the ProxyAPI and SmartProxy#version classes work as expected - instead of testing the ProxyAPI in the API functional controller.

This will mean we can do unit tests of the SmartProxy#version more cheaply than we can in a functional test, plus stubbing at a higher level will mean we avoid creating RestClient responses just for a functional test.